libraries/AP_NavEKF2/AP_NavEKF2_MagFusion.cpp

```diff
@@ -265,6 +265,26 @@ void NavEKF2_core::SelectMagFusion()
         }
     }
 
+    // If the final yaw reset has been performed and the state variances are sufficiently low
+    // record that the earth field has been learned.
+    bool earthMagFieldConverged = false;
+    if (!magFieldLearned && finalInflightMagInit) {
+        earthMagFieldConverged = (P[16][16] < sq(0.01f)) && (P[17][17] < sq(0.01f)) && (P[18][18] < sq(0.01f));
+    }
+    if (magFieldLearned || earthMagFieldConverged) {
```

Comment:
And what is the problem with that? There's no else, so it won't change the _magFieldLearned_ value. Actually if _magFieldLearned_ is true, then _earthMagFieldConverged_ is false.